### DISCUSSION TITLE
How can we implement matrix logic in our own generator plugin ?

### DISCUSSION THREAD

**Discussion by gabrielfsousa** [argoproj/argo-cd#18701] [General]
How can we implement matrix logic in our own generator plugin ?

**Comment by rumstead** [comment]
Why would you need to implement a matrix in your plugin vs using the built-in matrix and accepting the matrixed parameters as input? https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-Plugin/#with-matrix-and-pull-request-example

**Comment by gabrielfsousa** [comment]
Thanks for the reply,
I'm using right now
![image](https://github.com/argoproj/argo-cd/assets/722744/5e911de6-6d25-45af-b9ca-acff194a664e)


i have 2 reasons why i would like to implement matrix inside gen plugin
1.  add more logic on it, that is impossible to do it with vanilla matrix 
2. there's a bug in argocd (since ever i already reported) , **path / pathParamPrefix** only works in first generator, if we use it on second, the **.targets.path.segments** will not be replaced
